### PR TITLE
Initialize config and logging before main() starts to execute

### LINKED ISSUE #1475: Glusterd2--version displays some logs along with it
### Observed behavior

```
[root@dhcp35-30 ~]# glusterd2 --version
INFO[0000] Registering block provider                    name=gluster-block
INFO[0000] Registered block provider                     name=gluster-block
glusterd version: v6.0-dev.108.git79d9987
git SHA: 79d9987
go version: go1.11.2
go OS/arch: linux/amd64
```
### Information about the environment:

- Glusterd2 version used (e.g. v4.1.0 or master): 
- Operating system used: 
- Glusterd2 compiled from sources, as a package (rpm/deb), or container: 
- Using External ETCD: (yes/no, if yes ETCD version):
- If container, which container image: 
- Using kubernetes, openshift, or direct install: 
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside: 

### Other useful information

- glusterd2 config files from all nodes (default /etc/glusterd2/glusterd2.toml)
- glusterd2 log files from all nodes (default /var/log/glusterd2/glusterd2.log)
- ETCD configuration
- Contents of `uuid.toml` from all nodes (default /var/lib/glusterd2/uuid.toml)
- Output of `statedump` from any one of the node

### Useful commands

- To get glusterd2 version
    ```
    glusterd2 --version
    ```
- To get ETCD version
    ```
    etcd --version
    ```
- To get output of statedump
    ```
    curl http://glusterd2-IP:glusterd2-Port/statedump
    ```

### PR DESCRIPTION
**Pull request by oshankkumar**
…func

closes #1475

Signed-off-by: Oshank Kumar <okumar@redhat.com>

### CODE REVIEW

**Comment by oshankkumar** [comment]
retest this please

**Comment by oshankkumar** [comment]
retest this please

**Comment by oshankkumar** [comment]
retest this please

File: plugins/blockvolume/blockprovider/blockvolume_provider.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func RegisterBlockProvider(name string, f ProviderFunc) {
 		log.WithField("name", name).Error("failed to register block provider, provider already exist")
 		return
 	}
-	log.WithField("name", name).Infof("Registered block provider")
+	log.WithField("name", name).Debug("Registered block provider")
```
Comment (atinmu) [inline plugins/blockvolume/blockprovider/blockvolume_provider.go]: Does this mean that when we execute 'glusterd2 --version' all the logs will be captured in the console and we're fixing it by just changing the log level to debug. IOW, if we don't redirect the logs to the console, https://github.com/gluster/glusterd2/issues/1475 isn't a problem at all?

Comment (oshankkumar) [reply plugins/blockvolume/blockprovider/blockvolume_provider.go]: logs are getting initialized in main func, and all block providers got initialized in init(). For this, we need to init logging before main getting start to execute. @aravindavk @Madhu-1 can you please suggest, should we init conf and logging before main gets start to execute?

File: glusterd2/conf/config.go [hunk 2]
```diff
@@ -145,32 +150,32 @@ func initConfig() error {
...
+	// If custom configuration is passed use it, if not try to use defaults
+	err := config.ReadInConfig()
+	if err == nil {
+		return setDefaults()
 	}
 
-	// Finally initialize missing config with defaults
-	err := setDefaults()
+	if _, ok := err.(config.ConfigFileNotFoundError); ok {
+		return setDefaults()
```
[the remark this answers is not in the document]
Comment (oshankkumar) [reply glusterd2/conf/config.go]: done

**Pull request merged**